dkist/dataset/loader.py

```diff
@@ -14,6 +19,9 @@
     from asdf import ValidationError
 
 
+ASDF_FILENAME_PATTERN = r"^(?P<instrument>[A-Z-]+)_L1_(?P<timestamp>\d{8}T\d{6})_(?P<datasetid>[A-Z]+)(?P<suffix>_user_tools|_metadata)?.asdf$"
```

Comment:
Is the second `-` in the instrument group deliberate?

Reply:
yes, it's a literal `-` so it's matching anything from `A-Z` or `-`.